### ISSUE TITLE
External Storage available user GUI issue

### ISSUE THREAD

**Issue by iliyasbasir** [owncloud/core#13781]
### Expected behavior

I'm setting up some external storage to an smb share to be restricted by user/group. In testing I've just added it to my account and one other persons. I see the two users in mount.json defined by their active directory uid. 

When I go back to the admin page I see the mounts listed but the available users box is empty instead of showing the two grey buttons with my name and the other guy's. I've then added a third person. Only they show up in the web interface but all 3 users show in mount.json. If I reload the admin page again the box is empty. 

The behavior is the same on 6.0.6 and 7.0.4. And no buttons for users means no little x's to click on to remove a user. I can do it in the mount.json file but it'd be better not to. 

### Actual behavior
Should be showing name at Available for box with x in fron of username in EXTERNAL STOREGE section 

### Steps to reproduce

1) Add SMB add user  pointed to SMB ...but i'm unable reproduce issue 
2) My system behave as design

### Server configuration

Operating system:centos 6.6 

Web server:Apache 2.2.15

Database:

PHP version:5.5.20

ownCloud version:7.0.4

Storage backend:

Client configuration

Client version:

Operating system:

OS language:

Installation path of client:

### Logs

Logs /image can be found at s3 under support/github-issues/core/13781
@MorrisJobke

**Comment by PVince81** [comment]
Is LDAP involved ?

**Comment by LukasReschke** [comment]
We need the `mount.json`, post an anonymised copy of it. (replace username / password / host)

**Comment by iliyasbasir** [comment]
@PVince81 : Yes, I placed some detail with log regarding LDAP
@LukasReschke : I'll ask customer if he can provide mount.json

**Comment by cdamken** [comment]
@PVince81 @blizzz  @LukasReschke  Reading the Logs, the last 2 lines paragraph shows the solution/ workaround :

>I used Firefox 35 on Windows 7. ownCloud is version 7.0.4 running on Centos 6.6 with Apache 2.2.15 and PHP 5.5.20. The accounts are in Active Directory. I've attached a screen shot from Firefox showing two mounts with the issue. I've blurred the server address and username in case you need to post to a public github issue. If you do, please obscure uids, gids, usernames and display names the log entries above. Thanks.

>Looking at the log entries I see ldap queries looking at the displayname and mail attributes matching on uid. I think this might be related. It is doing an ldap query for displayname or mail because these are the two attributes in our search filter, but it is pulling the users internal oc username as the value to search which for ldap users will be their uid. This is probably the external storage system saying to users_ldap to search for this and giving the internal username (uid) and users_ldap is just running the filter it's been given, unaware that it has been supplied with a uid as a search value. Of course it finds no results. A workaround would be to include the uid attribute in our search filter in addition to displayname and mail. It will have no material impact on any of the search interfaces within the application. And if this is what you guys determine to the supported configuration then we have no problem with that. It probably just needs a note in the ldap documentation to point this out.

@blizzz  Maybe the patch S3-ownCloud\Shared\owncloud\support\Patches\ldap-2nddn-7.0.4.patch that you, with a few tweaks, could be a good implementation.

**Comment by blizzz** [comment]
Hm, actually I believe to remember that was tested, seen and also fixed in the original PR https://github.com/owncloud/core/pull/10576 (or at least the back-then stable7 or master port). It is supposed to work correctly without adding the corresponding user name attribute to the search ones.

Ok, after looking it up it seems it only landed to back-then master, neither stable6 nor 7. It uses the appframework which has not been part for OC 6 iirc. In that case, the workaround is favourable…

cc @butonic

**Comment by cdamken** [comment]
@blizzz Thanks

**Comment by butonic** [comment]
@blizzz then master: https://github.com/owncloud/core/pull/8507 
stable7: https://github.com/owncloud/core/pull/10558
stable6: https://github.com/owncloud/core/pull/10576

Follow up work regarding flickering of users went into
master: https://github.com/owncloud/core/pull/11552
stable7: https://github.com/owncloud/core/pull/11550

Also, @iliyasbasir can't reproduce this. We need logs / mount.json is @LukasReschke pointed out

**Comment by iliyasbasir** [comment]
@MorrisJobke @butonic @LukasReschke : I did placed logs / mount.json  can be found at s3 under support/github-issues/core/13781

**Comment by LukasReschke** [comment]
Public bug tracker – so why not make the anonymized mount.json public as well? ;-)

**Comment by bboule** [comment]
@iliyasbasir We need to make sure we have the information that @butonic is looking for!!

**Comment by craigpg** [comment]
Are there next steps?  Do we have requested info?  Looks like logs / mount.json are available on s3.

**Comment by iliyasbasir** [comment]
@craigpg @PVince81 @LukasReschke : I had impression that this issue should resolved on 7.0.5 but this release did not fixed  issue 
can you please help Customer been asking for fix

**Comment by PVince81** [comment]
The attached mount.json only contains a single user. Is it from the moment where you overwrote it with the new user ? Can you try assigning it again to two users, then refresh the page, and see if they appear ? If not, please check the Javascript console for errors.

After adding users in the "Applicable" field, you might need to click outside and wait a second until the spinner appears. Only then it is properly saved. Let me know if that worked. (maybe we need to add a warning if the page is left with unsaved changes)

**Comment by bboule** [comment]
@iliyasbasir can you take a look at @PVince81 comments and see if you can address the issue based on his suggestion?

**Comment by iliyasbasir** [comment]
@PVince81 @bboule , Yes i forward Vincent recommendation to customer

**Comment by iliyasbasir** [comment]
@PVince81 Here is note from customer 

```
Hi Iliyas,

The issue isn't whether changes get saved or not. I can add a user or group to external storage just fine. That change is saved almost instantly and is reflected in the mounts.json file and in the accounts of the users in question. They can see the external storage just fine. And the admin page shows the users and/or groups that have just been added just fine.

But if I as the admin leave the admin page and come back, there is no list of users displayed, just an empty white box. I can add additional users or groups to the mount and those changes are saved correctly. The mounts.json file updates and the mount is available to the new user account straight away as expected. And those users display in the admin page as though they are the only ones linked to that external storage mount even though they aren't. Reload the page or go where else and come back and it's empty again.

The form in the admin page just doesn't show any users or groups upon reload. It's a display issue. The functionality is working and users have access to the storage that is configured. It's just that the form is not populating with the existing list of users and groups linked to the external storage.

We had a screen sharing session where I demonstrated this for Bob so he saw it in action. Issue is exactly as described in 
https://github.com/owncloud/core/issues/15800

 It seems that there is a fix that has been backported to 7 but Karl has said it should be in 7.0.7 not 7.0.6 (https://github.com/owncloud/core/pull/15833#issuecomment-95577276). Could you see if he could change his mind so we can get this feature fixed with the next release rather than wait longer? Otherwise we will need to backport ourselves which is something we'd really like to avoid.

Regards,
```

**Comment by MorrisJobke** [comment]
@karlitschek This is the reference of the comment from https://github.com/owncloud/core/pull/15833#issuecomment-98471959 ;)

**Comment by MorrisJobke** [comment]
This will be fixed in 7.0.6 - #15833 was merged

**Comment by iliyasbasir** [comment]
@MorrisJobke : I did not get your comment "This will be fixed in 7.0.6 - #15833 was merged"  we are at oC 8.0.3

My apology in advanced if i did not get your wording right

**Comment by MorrisJobke** [comment]
You opened this ticket for an ownCloud 7 instance. So I commented for the 7.0.x version. The fix is already present in 8.0.x and 8.1.x
